src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -73,7 +75,12 @@ public bool BrowserVisibility
             set { browserVisibility = value; RaisePropertyChanged("BrowserVisibility"); }
         }
 
-        private string searchText;
+        internal int searchDelayTimeout = 150;
+        private bool useDebouncer = true; // replace with feature flag check
+        private ActionDebouncer searchDebouncer = new ActionDebouncer(Dispatcher.CurrentDispatcher);
```

Comment:
so you have to be a bit careful using method of retrieving the dispatcher - if this code is called from a non ui thread the dispatcher will not be the WPF dispatcher.

That seems unlikely in most scenarios. If you have access to the DynamoViewModel here there is a property UIDispatcher that should always be correct, as its set from the WPF main window's dispatcher.

Reply:
Yeah, that Dispatcher retrieval issue was the original reason for having the initialization inside the SearchText setter, with the assumption that the setter would only be called from the UI thread.

The DynamoViewModel UIDispatcher seems to be set by the View it belongs to after the SearchViewModel is constructed, so that requires the debouncer to be constructed lazily at some point in the future when the UIDispatcher has been set, rather than during the SearchViewModel construction.

So then I think we're back to some variation on the first version with a null check inside of the SearchText-setter?

Reply:
`Yeah, that Dispatcher retrieval issue was the original reason for having the initialization inside the SearchText setter, with the assumption that the setter would only be called from the UI thread.` ah ok.

I'm not sure that's a strong assumption. It's a public method on a public class running in an application full of extensions and packages.

For now, what about just wrapping this call in a try catch and logging, it seems unlikely, but possible it could be a non-UI thread dispatcher and, in that case, it might crash when the search results get modified from the wrong thread - unless this only causes binding updates anyhow that will get marshalled to the UI thread by WPF?

Reply:
That's fair, it probably was the wrong assumption on my part.

1. Alright, I went back to the lazy initialization.
2. I wrapped the call as suggested. I don't think the debouncer can ever use anything other than UIDispatcher now, unless someone sets the wrong dispatcher on the DynamoViewModel, but then I think bigger issues are in store.

I did try setting up a test where the Dispatcher gets set from another thread, and the SearchText is set from that same thread, and then again set from inside. It all works, even with the wrong Dispatcher. 

Where it does fail is if I try to set the Text property directly on the control, which throws an exception due to using the wrong Dispatcher, but I don't think that has anything to do with the debouncer dispatcher.